odoo/modules/registry.py

```diff
@@ -901,7 +901,7 @@ def signal_changes(self):
             with closing(self.cursor()) as cr:
                 for cache_name in self.cache_invalidated:
                     cr.execute("select nextval(%s)", [f'base_cache_signaling_{cache_name}'])
-                    self.cache_sequences[cache_name] = cr.fetchone()[0]
+                    self.cache_sequences[cache_name] += 1
```

Comment:
Yes, you are right!
Both the old `self.cache_sequences[cache_name] = cr.fetchone()[0]` and my new `self.cache_sequences[cache_name] += 1` are not atomic. In multi-thread mode, the registry may have out-dated sequence number while `check_signaling` which leads to redundant registry reload / cache invalidation.
But because the race is really rare, I think the performance drop down is acceptable.